doc/manual/src/command-ref/nix-shell.md

```diff
@@ -297,3 +297,9 @@ with import <nixpkgs> {};
 
 runCommand "dummy" { buildInputs = [ python pythonPackages.prettytable ]; } ""
 ```
+
+Aside from the very first line, which is a directive to the operating system,
+the `#! nix-shell` lines do not need to be at the beginning of the file. This
+can be useful for working around syntactic restrictions in languages like
+ECMAScript, Erlang, PHP, and Ruby, where the `#! nix-shell` lines must be
```

Comment:
It may be correct, but do people say ECMAScript?

```suggestion
JavaScript, Erlang, PHP, and Ruby, where the `#! nix-shell` lines must be
```

Reply:
Noted. I'll change it if required, but my time working with ECMAScript's architects has convinced me that the trademarked "JavaScript" moniker should be avoided when referring to the more generic usage of the language outside browser contexts. Thanks for the review!